src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_MediaPlayerElement.cs

```diff
@@ -0,0 +1,140 @@
+{
+	[TestClass]
+	[RunsOnUIThread]
+	public class Given_MediaPlayerElement
+	{
+		[TestMethod]
+		public void When_MediaPlayerElement_Not_SetSource_Property_Value()
+		{
+			var entity = new MediaPlayerElement();
+			entity.SetMediaPlayer(new Windows.Media.Playback.MediaPlayer());
+			entity.MediaPlayer.Play();
+			entity.GetValue(MediaPlayerElement.SourceProperty);
+
+			Assert.IsNull(entity.Source);
+		}
+
+		[TestMethod]
+		public void When_MediaPlayerElement_SetSource_Property_Value()
+		{
+			var entity = new MediaPlayerElement();
+			entity.AutoPlay = true;
+			entity.Source = Windows.Media.Core.MediaSource.CreateFromUri(new Uri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_5MB.mp4"));
+			entity.GetValue(MediaPlayerElement.SourceProperty);
+			entity.Should().NotBeNull();
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_Added_In_Opening()
+		{
+			MediaPlayerElement entity = await StartPlayerAsync();
+
+			entity.MediaPlayer.Play();
+
+			var mediaTransportControls = entity.TransportControls as MediaTransportControls;
+			Assert.IsNotNull(mediaTransportControls);
+
+			var mediaPlayer = entity.MediaPlayer as Windows.Media.Playback.MediaPlayer;
+			Assert.IsNotNull(mediaPlayer);
+		}
+
+		[TestMethod]
+		public void When_MediaPlayerElement_Dependency_Property_Value()
+		{
+			var entity = new MediaPlayerElement();
+			entity.AutoPlay = true;
+			entity.Source = Windows.Media.Core.MediaSource.CreateFromUri(new Uri("https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/720/Big_Buck_Bunny_720_10s_5MB.mp4"));
+			entity.ApplyTemplate();
+			entity.SetMediaPlayer(new Windows.Media.Playback.MediaPlayer());
+			entity.MediaPlayer.Play();
+
+			entity.GetValue(MediaPlayerElement.SourceProperty);
+			entity.Should().NotBeNull();
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_SetIsFullWindow_Check_Fullscreen()
+		{
+
+			MediaPlayerElement entity = await StartPlayerAsync();
+			try
+			{
+
+				entity.IsFullWindow = true;
+
+				Thread.Sleep(3000);
+				entity.GetValue(MediaPlayerElement.IsFullWindowProperty);
+				entity.Should().NotBeNull();
+			}
+			finally
+			{
+				entity.IsFullWindow = false;
+			}
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_SetSource_Check_Play()
+		{
+			MediaPlayerElement entity = await StartPlayerAsync();
+
+			entity.MediaPlayer.Play();
+
+			Thread.Sleep(3000);
+			Assert.IsTrue(entity.MediaPlayer.PlaybackSession.IsPlaying);
+		}
+
+		[TestMethod]
+		public async Task When_MediaPlayerElement_SetSource_Check_PausePlayStop()
+		{
+			MediaPlayerElement entity = await StartPlayerAsync();
+
+			entity.MediaPlayer.Play();
+			Thread.Sleep(3000);
+			Thread.Yield();
+
+			Assert.AreEqual(entity.MediaPlayer.PlaybackSession.PlaybackState, MediaPlaybackState.Playing);
+
+			// step 2: Test Stop
+			entity.MediaPlayer.Stop();
+			Assert.AreEqual(entity.MediaPlayer.PlaybackSession.Position, TimeSpan.Zero);
```

Comment:
might want to add a `WaitForIdle()` between action and validation for response

Reply:
Done.